### ISSUE TITLE
How can I remove replica in ClickHouse-Keeper when `REPLICA_ALREADY_EXISTS`?

### ISSUE THREAD

**Issue by LGDHuaOPER** [ClickHouse/ClickHouse#47920]
ClickHouse version is `23.1.3.5`.

When I create table use:

```sql
CREATE TABLE db.log_local ON CLUSTER all (
	`user` String,
    `x_request_id` String,
    `col1` UInt32
)
ENGINE = ReplicatedMergeTree
ORDER BY (`user`, xxHash32(`x_request_id`))
```

But it raise error like:

```text
Received exception from server (version 23.1.3):
Code: 253. DB::Exception: Received from localhost:9000. DB::Exception: There was an error on [1.2.3.4:9000]: Code: 253. DB::Exception: Replica /clickhouse/tables/clickhouse_shard_num_1/db/log_local/replicas/replica_num_2 already exists. (REPLICA_ALREADY_EXISTS) (version 23.1.3.5 (official build)). Stack trace:

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0xddb0df5 in /usr/bin/clickhouse
1. DB::DDLQueryStatusSource::generate() @ 0x138056a2 in /usr/bin/clickhouse
2. DB::ISource::tryGenerate() @ 0x146c1575 in /usr/bin/clickhouse
3. DB::ISource::work() @ 0x146c10e6 in /usr/bin/clickhouse
4. DB::ExecutionThreadContext::executeTask() @ 0x146db366 in /usr/bin/clickhouse
5. DB::PipelineExecutor::executeStepImpl(unsigned long, std::__1::atomic<bool>*) @ 0x146d047c in /usr/bin/clickhouse
6. DB::PipelineExecutor::executeImpl(unsigned long) @ 0x146ceeb9 in /usr/bin/clickhouse
7. DB::PipelineExecutor::execute(unsigned long) @ 0x146cec00 in /usr/bin/clickhouse
8. ? @ 0x146dedff in /usr/bin/clickhouse
9. ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>) @ 0xde7de76 in /usr/bin/clickhouse
10. ? @ 0xde831a1 in /usr/bin/clickhouse
11. ? @ 0x7f9580950609 in ?
12. __clone @ 0x7f9580875133 in ?
. (REPLICA_ALREADY_EXISTS)
```

How can I remove the replica in ClickHouse-Keeper? Can you give me some help or docs? Or there are some commands can do this?

Thanks!

**Comment by tavplubix** [comment]
First of all, you need to find out why it already exists. Probably you use not unique paths in ZooKeeper. In this case, you should not remove anything. See also https://github.com/ClickHouse/ClickHouse/issues/20243

Then you can use https://clickhouse.com/docs/en/sql-reference/statements/system#drop-replica

**Comment by sbalasa** [comment]
---

I encountered this with one of my tables, I applied the following resolution:

1. Manually instantiate the table on all nodes, excluding the `ON CLUSTER` clause during table creation.
2. If the error "REPLICA_ALREADY_EXISTS" is thrown, modify the ZooKeeper path slightly by appending or altering a segment of the path to ensure uniqueness.
3. Once the table is successfully created across nodes, execute the command:

`system sync replica on cluster '{cluster}' <db_name>.<table_name> strict ;`

This command ensures synchronization across replicas and should rectify the `"REPLICA_ALREADY_EXISTS"` error.
